src/gpl/test/gpl_aux.py

```diff
@@ -0,0 +1,195 @@
+    elif initial_place_max_fanout != None:
+        utl.error(utl.GPL, 505, "initial_place_max_fanout must be a positive integer")
+
+    uniform_mode = False
+
+    if density != None:
+        target_density = density
+    else: 
+        target_density = DEFAULT_TARGET_DENSITY
+
+    if target_density == "uniform":
+        uniform_mode = True
+    elif is_pos_float(target_density):
+        gpl.setTargetDensity(target_density)
+
+    gpl.setUniformTargetDensityMode(uniform_mode)
+
+    if is_pos_float(routability_max_density):
+        gpl.setRoutabilityMaxDensity(routability_max_density)
+
+    if is_pos_float(min_phi_coef):
+        gpl.setMinPhiCoef(min_phi_coef)
+
+    if is_pos_float(max_phi_coef):
+        gpl.setMaxPhiCoef(max_phi_coef)
+
+    if is_pos_float(init_density_penalty):
+        gpl.setInitDensityPenalityFactor(init_density_penalty)
+
+    if is_pos_float(init_wirelength_coef):
+        gpl.setInitWireLengthCoef(init_wirelength_coef)
+
+    if is_pos_float(reference_hpwl):
+        gpl.setReferenceHpwl(reference_hpwl)
+
+    if is_pos_int(bin_grid_count):
+        gpl.setBinGridCntX(bin_grid_count)
+        gpl.setBinGridCntY(bin_grid_count)
+
+    if is_pos_float(overflow):
+        gpl.setTargetOverflow(overflow)
+
+    if is_pos_float(routability_check_overflow):
+        gpl.setRoutabilityCheckOverflow(routability_check_overflow)
+
+    if is_pos_float(routability_max_bloat_iter):
+        gpl.setRoutabilityMaxBloatIter(routability_max_bloat_iter)
+
+    if is_pos_float(routability_max_inflation_iter):
+        gpl.setRoutabilityMaxInflationIter(routability_max_inflation_iter)
+
+    if is_pos_float(routability_target_rc_metric):
+        gpl.setRoutabilityTargetRcMetric(routability_target_rc_metric)
+
+    if is_pos_float(routability_inflation_ratio_coef):
+        gpl.setRoutabilityInflationRatioCoef(routability_inflation_ratio_coef)
+
+    if is_pos_float(routability_max_inflation_ratio):
+        gpl.setRoutabilityMaxInflationRatio(routability_max_inflation_ratio)
+
+    if routability_rc_coefficients != None:
+        k1, k2, k3, k4 = routability_rc_coefficients
+        gpl.setRoutabilityRcCoefficients(k1, k2, k3, k4)
+
+    # Now here is a section that is marked "temp code" in the tcl file
+
+    if is_pos_int(pad_left):
+        gpl.setPadLeft(pad_left)
+
+    if is_pos_int(pad_right):
+        gpl.setPadRight(pad_right)
+
+    if len(design.getBlock().getRows()) > 0:  # db_has_rows
+        if incremental:
+            gpl.doIncrementalPlace()
+        else:
+            gpl.doInitialPlace()
+            if not skip_nesterov_place:
+                gpl.doNesterovPlace()
+
+        gpl.reset()
+        gpl.setDb(design.getTech().getDB())
+        gpl.setLogger(design.getLogger())
+        gpl.setGlobalRouter(design.getGlobalRouter())
+        # This will have to wait until rsz is Python wrapped
+        #rz = design.getResizer()
+        #gpl.setResizer(rz)
```

Comment:
Does this mean that something doesn't currently work in Python?

Reply:
No, this just means we haven't yet wrapped rsz in Python. To get the resizer before we wrap it in Python, we would have to update design.evalTclString("...") to return objects. I did update it in this PR to return strings, because that is needed for some conditional tests that are specific to STA, but I did not want to start getting objects back from tcl to Python.